web/packages/teleport/src/lib/tdp/client.ts

```diff
@@ -611,9 +611,12 @@ export default class Client extends EventEmitterWebAuthnSender {
   }
 
   sendKeyboardInput(code: string, state: ButtonState) {
-    // Only send message if key is recognized, otherwise do nothing.
-    const msg = this.codec.encodeKeyboardInput(code, state);
-    if (msg) this.send(msg);
+    const msgs = this.codec.encodeKeyboardInput(code, state);
+    if (msgs) {
```

Comment:
If you make `encodeKeyboardInput` always return an array (which can be empty) then the typing becomes simpler and this `if` can go away. (`forEach` on an empty array is a no-op)

Reply:
adf8c2fbf9eb4d7a2813aa4d41d75048ae7e18f3